pkg/util/test/util.go

```diff
@@ -643,6 +644,71 @@ func validateImageTag(tag, namespace string, listOptions map[string]string) erro
 	return nil
 }
 
+func validateMonitoring(cluster *corev1alpha1.StorageCluster, timeout, interval time.Duration) error {
+	if cluster.Spec.Monitoring != nil &&
+		(*cluster.Spec.Monitoring.EnableMetrics ||
+			(cluster.Spec.Monitoring.Prometheus != nil && cluster.Spec.Monitoring.Prometheus.ExportMetrics)) {
+		if cluster.Spec.Monitoring.Prometheus != nil && cluster.Spec.Monitoring.Prometheus.Enabled {
+			dep := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "prometheus-operator",
+					Namespace: cluster.Namespace,
+				},
+			}
+			if err := appops.Instance().ValidateDeployment(&dep, timeout, interval); err != nil {
+				return err
+			}
+
+			st := appsv1.StatefulSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "prometheus-prometheus",
```

Comment:
Name should be "px-prometheus-prometheus" or 
`component.PrometheusInstanceName + "-prometheus"`